lumibot/entities/data.py

```diff
@@ -519,7 +519,7 @@ def get_bars(self, dt, length=1, timestep=MIN_TIMESTEP, timeshift=0):
         if data is None:
             return None
 
-        df = pd.DataFrame(data).set_index("datetime")
+        df = pd.DataFrame(data).assign(datetime=lambda df: pd.to_datetime(df['datetime'])).set_index('datetime')
```

Comment:
The lambda part is unnecessary as you can do the to_datetime call directly:
```python
pd.DataFrame(data).assign(datetime=pd.to_datetime(df['datetime']))
```

However, I believe that the 'datetime' column as already been converted to a pd.datetime series.  Did you encounter an error somewhere that this fixed?

Reply:
The lambda works but not without because df isn't defined yet.  I looked around and this lambda style seems pretty common.

Yeah, this is the first bug I encountered running backtest with Polygon.  I just patched it and moved on but it had something to do with resampling from minute bars probably when some were missing.  I will recheck to make sure it reproduces and see if I can come up with a test.